src/command/verify/verify.c

```diff
@@ -1841,16 +2044,38 @@ verifyProcess(const bool verboseText)
                 // ??? Need to do the final reconciliation - checking backup required WAL against, valid WAL
 
                 // Report results
-                resultStr = verifyRender(jobData.archiveIdResultList, jobData.backupResultList, verboseText);
+                resultStr = verifyRender(jobData.archiveIdResultList, jobData.backupResultList, verboseText, resultKv, errorList);
             }
-            else if (!backupLabelInvalid)
+            else if (!json && !backupLabelInvalid)
                 strCatZ(resultStr, "\n    no archives or backups exist in the repo");
 
             errorTotal += jobData.jobErrorTotal;
         }
 
+        if (json)
+        {
+            kvPut(resultKv, VERIFY_KEY_STANZA_VAR, VARSTR(cfgOptionStr(cfgOptStanza)));
+            kvPut(resultKv, KEY_STATUS_VAR, errorTotal > 0 ? VERIFY_KEY_STATUS_ERROR : VERIFY_KEY_STATUS_OK);
+
+            kvPut(resultKv, KEY_ERRORS_VAR, varNewVarLst(errorList));
+
+            strCat(result, jsonFromVar(varNewKv(resultKv)));
+        } else {
```

Comment:
```suggestion
        }
        else
        {
```
I don't know why, but upstream is using this style